### DISCUSSION TITLE
Understanding the training feature

### DISCUSSION THREAD

**Discussion by shnchh** [birdnet-team/BirdNET-Analyzer#446] [Q&A]
I would like to understand what happens when you use the append mode in training. I understand that birdNET concatenates its own predictions onto that of the trained linear classifier. As far as I can tell, when there is a species that exists in the linear classifier and my birdnet's labels, both will be outputted independently as predictions. As a result, I am getting strong predictions for a target species from my classifier on my noise. I would like to ask how to handle these duplicate predictions, and if general, if I am trying to train a model to recognise a subset of birdNET's species, but don't neccessarily have data for every one of them, how I should go about training the linear classifier with data I do have. This would be several hundred species, so would the linear classifier be even workable? I see from the few-shot transfer learning paper that the typical use cases only involve a few classes.

**Comment by kahst** [comment]
You're right, if you train a species that already is in BirdNET and then append the new classifier you'll see duplicate detections for this species. Yet, it's totally up to you how to handle that. Here are a few recommendations: 1) Ignore the original prediction and only trust your new one, or 2) Combine the scores and require both classifiers to score above a certain threshold. 

There's no cap in the numbers of species you can reliably have in the newly trained classifier. In our embeddings paper, we argue that we can reliably detect any audio class based on the embeddings and the new classifier treats each species independently. So, you could even have 1,000 or more classes in the new classifier, IF you have the data. The only constraint might be the model size, which (of course) increases with each new class (for 1,000 new species times 1,024 - the embeddings size - that would be ~1mio weights).

**Comment by Josef-Haupt** [comment]
3.) You could also weight each prediction, ie.: 0.4 * `base BirdNET score` + 0.6 * `appended BirdNET score`

**Comment by shnchh** [comment]
Thanks all! I am wondering if there is a way to feed my data directly to the existing birdNET linear classifier for these duplicate species?
